logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/Message.kt

```diff
@@ -60,6 +61,42 @@ sealed interface Message {
         val visibility: Visibility
     }
 
+    data class Ephemeral(
+        val expireAfterMillis: Long,
+        val selfDeletionStartDate: Long?,
+        val ephemeralMessage: Regular
+    ) : Standalone {
+
+        fun timeLeftForDeletion(): Long {
```

Comment:
Wouldn't it be better, instead of returning the remaining time, to return the time until which it should be displayed? Usually it shouldn't make any difference, but what if something is lagged/delayed, then it can be shown for longer that it should actually

Reply:
Yes, returning a date makes more sense.

Reply:
That also ties into my arguments about timers. We don't need a `flow<timeLeftInMiliseconds>, the UI can render a `CountDown(Clock.System.now() -  message.expirationDate)`, which it refreshes that at whatever interval it deems appropriate.